Makefile

```diff
@@ -67,4 +68,6 @@ smoke:
 integration:
 	CGO_ENABLED=1 ${PROXY} GOOS=${GOOS} GOARCH=${GOARCH} go build -ldflags '-s -w -X "main.Version=${VERSION}" -extldflags "-static"' -race -v -o bin/containerd-nydus-grpc ./cmd/containerd-nydus-grpc
 	$(SUDO) DOCKER_BUILDKIT=1 docker build -t nydus-snapshotter-e2e:0.1 -f integration/Dockerfile .
-	$(SUDO) docker run --privileged -v /root/.docker:/root/.docker -v `go env GOMODCACHE`:/go/pkg/mod -v `go env GOCACHE`:/root/.cache/go-build -v `pwd`:/nydus-snapshotter nydus-snapshotter-e2e:0.1
\ No newline at end of file
+	$(SUDO) docker run --rm --privileged -v /root/.docker:/root/.docker -v `go env GOMODCACHE`:/go/pkg/mod \
+	-v `go env GOCACHE`:/root/.cache/go-build -v `pwd`:/nydus-snapshotter \
+	-v /usr/src/linux-headers-${KERNEL_VER}:/usr/src/linux-headers-${KERNEL_VER} nydus-snapshotter-e2e:0.1
```

Comment:
why do we need kernel headers here?

Reply:
Because the test has to detect if the kernel is supporting erofs on-demand read